ads/google/doubleclick.js

```diff
@@ -265,3 +245,54 @@ function doubleClickWithGlade(global, data, gladeExperiment) {
 function getCorrelator(global) {
   return makeCorrelator(global.context.clientId, global.context.pageViewId);
 }
+
+function centerAd() {
+  const container = global.document.querySelector('#c');
+  setStyles(dev().assertElement(container), {
+    top: '50%',
+    left: '50%',
+    bottom: '',
+    right: '',
+    transform: 'translate(-50%, -50%)',
+  });
+}
+
+/**
+ * @param {Object} data
+ * @return {!string}
+ */
+export function getUrl(data) {
+  const fileExperimentConfig = {
+    21060540: 'gpt_sf_a.js',
+    21060541: 'gpt_sf_b.js',
+  };
+  // Note that reduce will return the first item that matches but it is
+  // expected that only one of the experiment ids will be present.
+  let expFilename;
+  (data['experimentId'] || '').split(',')
+  .forEach(val => expFilename = expFilename || fileExperimentConfig[val]);
+  return `https://www.googletagservices.com/tag/js/${expFilename || 'gpt.js'}`;
+}
+
+/**
+ * @param {Object} data
+ * @param {!string} url
+ * @param {number} experimentFraction
+ */
+export function whichGladeExperimentBranch(data, url, experimentFraction) {
+  if (data.useSameDomainRenderingUntilDeprecated != undefined
+    || data.multiSize || url.indexOf('sf_a') > -1 || url.indexOf('sf_b') > -1) {
+    doubleClickWithGpt(data, GladeExperiment.GLADE_OPT_OUT, url);
+  } else {
+    const dice = Math.random();
+    const href = global.context.location.href;
+    if ((href.indexOf('google_glade=0') > 0 || dice < experimentFraction)
+          && href.indexOf('google_glade=1') < 0) {
+      doubleClickWithGpt(data, GladeExperiment.GLADE_CONTROL, url);
+    } else {
+      const exp = (dice < 2 * experimentFraction) ?
```

Comment:
I think the former owners wrote this actually. But yes, this is because there are two branches, each getting 10%. This is the convention of how experiments have been written in GPT and carried over to here. I'm not a big fan of the inconsistency with our other experiment fractions where one fraction is divided among N branches.

Feel free to change as desired, but please let me know so I know whether I should expect any shifts. :)